src/Features/CSharp/Portable/Copilot/CSharpImplementNotImplementedExceptionFixProvider.cs

```diff
@@ -0,0 +1,175 @@
+        if (context.Diagnostics.Length == 0)
+        {
+            return;
+        }
+
+        var document = context.Document;
+        var cancellationToken = context.CancellationToken;
+
+        if (document.GetLanguageService<ICopilotOptionsService>() is not { } optionsService ||
+            await optionsService.IsImplementNotImplementedExceptionEnabledAsync().ConfigureAwait(false) is false)
+        {
+            return;
+        }
+
+        if (document.GetLanguageService<ICopilotCodeAnalysisService>() is not { } copilotService ||
+            await copilotService.IsAvailableAsync(cancellationToken).ConfigureAwait(false) is false)
+        {
+            return;
+        }
+
+        var throwNode = context.Diagnostics[0].Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+
+        // Preliminary analysis before registering fix
+        var methodOrProperty = throwNode.FirstAncestorOrSelf<MemberDeclarationSyntax>();
+        if (methodOrProperty is BasePropertyDeclarationSyntax or BaseMethodDeclarationSyntax)
+        {
+            var fix = DocumentChangeAction.New(
+                CSharpAnalyzersResources.Implement_with_Copilot,
+                async (_, cancellationToken) => await GetDocumentUpdater(context, null)(cancellationToken).ConfigureAwait(false),
+                (_, _) => Task.FromResult(context.Document),
+                nameof(CSharpAnalyzersResources.Implement_with_Copilot));
+            context.RegisterCodeFix(fix, context.Diagnostics[0]);
+        }
+    }
+
+    private static async Task<ImmutableArray<ReferencedSymbol>> FindReferencesAsync(Document document, ISymbol symbol, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        var progress = new StreamingProgressCollector();
+        await SymbolFinder.FindReferencesAsync(
+            symbol, document.Project.Solution, progress, documents: null,
+            FindReferencesSearchOptions.Default, cancellationToken).ConfigureAwait(false);
+
+        return progress.GetReferencedSymbols();
+    }
+
+    protected override async Task FixAllAsync(
+        Document document, ImmutableArray<Diagnostic> diagnostics,
+        SyntaxEditor editor, CancellationToken cancellationToken)
+    {
+        foreach (var diagnostic in diagnostics)
+        {
+            await FixOneAsync(editor, document, diagnostic, cancellationToken).ConfigureAwait(false);
+        }
+
+        var changedRoot = editor.GetChangedRoot();
+
+        // Get the language specific rule for formatting this construct and call into the
+        // formatter to explicitly format things. Note: all we will format is the new
+        // node as that's the only node that has the appropriate annotation on it.
+        var formattingOptions = await document.GetSyntaxFormattingOptionsAsync(cancellationToken).ConfigureAwait(false);
+        var rules = ImmutableArray.Create(MultiLineConditionalExpressionFormattingRule.Instance);
```

Comment:
still don't know why this is needed.  why are we specializeding conditional expression formatting?